packages/online_scrap_app/Classes/Controller/BookingsController.php

```diff
@@ -159,18 +159,16 @@ public function createBookingAction(Bookings $newBookings)
         $bookingId=trim($bookingId);
         var_dump($bookingId);
 
-        // filling booking details with view cart parameter
-        $newbookingDetails = new BookingDetails();
+        /** @var CartView $cart */
         foreach ($cartView as $cart) {
-            $category= new Category();
-            $category=$cart->getCategory();
-            $newbookingDetails->addCategory($category);
+            // filling booking details with view cart parameter
+            $newbookingDetails = new BookingDetails();
```

Comment:
By looking into that again, my change was probably wrong... In case you want to merge(!) all categories of `$cart` to `$newbookingDetails`
* `$newbookingDetails` again should be before the `foreach` loop
* inside the loop there might be a new loop assigning each single `$cart` category to `$newbookingDetails`, e.g.

```
foreach ($cartView as $cart) {
    ...
    foreach ($cart->getCategory() as $cartCategory) {
        $newbookingDetails->addCategory($cartCategory);
    }
   ...
}
```

Reply:
I case you need further support, please let me know what you actually tried to achieve...